### PR TITLE
fix label in back button

### PR DESCRIPTION
**Pull request by FariasBlu**
## Description

This PR fix label in back button

Fixes # (issue)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Screenshots (if appropriate):

**Before**

https://github.com/user-attachments/assets/9a43e4b6-e830-4321-8d76-4a605fecd1bd

**After**

https://github.com/user-attachments/assets/c1dfb476-9ac3-4611-8e63-d40d41574374

**Navigation in other flows**

https://github.com/user-attachments/assets/b30d5aae-7ed1-4b43-bec6-ebc43e2fcf5c

## Checklist:

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) document
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] All new and existing tests passed

### CODE REVIEW

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift [hunk 1]
```diff
@@ -73,24 +70,21 @@ public extension OceanNavigationBar {
         navBarAppearance.shadowColor = nil
         navBarAppearance.setBackIndicatorImage(navigationBackImage, transitionMaskImage: navigationBackImage)
 
-        if navigationShadow {
```
Comment (vromeiro) [inline OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:76]: volta com o navigationShadow

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift [hunk 2]
```diff
@@ -73,24 +70,21 @@ public extension OceanNavigationBar {
...
-        
-        self.navigationController?.navigationBar.setNeedsLayout()
-        self.navigationController?.navigationBar.layoutIfNeeded()
-        self.navigationController?.navigationBar.setNeedsDisplay()
+
+        updateTitle(newTitle: navigationTitle)
     }
     
     func updateTitle(newTitle: String) {
-        navigationController?.title = newTitle
```
Comment (vromeiro) [inline OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:93]: precisa remover essa linha?

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift [hunk 3]
```diff
@@ -112,7 +106,6 @@ public extension OceanNavigationBar {
     }
     
     func showBackButton(animated: Bool = true) {
-        navigationItem.leftBarButtonItem = nil
```
Comment (vromeiro) [inline OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:115]: precisa remover essa linha?

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift [hunk 4]
```diff
@@ -73,24 +70,21 @@ public extension OceanNavigationBar {
...
         navigationController?.navigationBar.standardAppearance = navBarAppearance
         navigationController?.navigationBar.compactAppearance = navBarAppearance
         navigationController?.navigationBar.scrollEdgeAppearance = navBarAppearance
         navigationController?.navigationBar.prefersLargeTitles = navigationLargeTitle
-        navigationItem.backButtonTitle = navigationBackButtonTitle
-        self.navigationController?.navigationBar.topItem?.backButtonTitle = navigationBackButtonTitle
-        
-        self.navigationController?.navigationBar.setNeedsLayout()
-        self.navigationController?.navigationBar.layoutIfNeeded()
-        self.navigationController?.navigationBar.setNeedsDisplay()
```
Comment (acassiovilasboas) [inline OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:89]: Celso, isso aqui afeta? Se nao afetar, deixa eles pq tem configuracoes da navigation controller nas linhas de cima.

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift [hunk 5]
```diff
@@ -49,9 +49,6 @@ public extension OceanNavigationBar {
     }
     
     func setupNavigation() {
-        navigationController?.navigationBar.tintColor = navigationTintColor
```
Comment (acassiovilasboas) [inline OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:52]: Essa configuracao aplica cor nos itens, ou só no texto?

**Review by acassiovilasboas — APPROVED** [review]

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift (hunk 2 above)
Comment (FariasBlu) [reply OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:93]: Baseado nesse [artigo](https://sarunw.com/posts/how-to-remove-text-from-uinavigationbar-back-button/?utm_source=chatgpt.com), precisamos

image

Comment (vromeiro) [reply OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:93]: hmm, é q pode acabar nao exibindo o title na navigation controller em alguns fluxos
se tiver essa linha o bug continua certo?

Comment (FariasBlu) [reply OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:93]: Com essa linha o bug continua. 

Fiz o teste navegando por outros fluxos, vou anexar o vídeo aqui na PR também

**Review by vromeiro — APPROVED** [review]

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift (hunk 1 above)
Comment (FariasBlu) [reply OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift]: Returned

File: OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift (hunk 5 above)
Comment (FariasBlu) [reply OceanComponents/Classes/Components UIKit/NavigationBar/Ocean+NavigationBar.swift:52]: Aplica a cor nos ícones e no texto

**Pull request merged**